roles/azure_manage_resource_group/tasks/delete.yml

```diff
@@ -16,13 +16,12 @@
         resource_group: "{{ azure_manage_resource_group_name }}"
       with_items: "{{ result.locks }}"
 
```

Comment:
Hi Nir,

The original wait_for 10 seconds was because it sometimes take a bit for a the lock to go away after deleting it right?  I think it would be better to do the retry on a new lock_info task inside the above block.  This way we wait for the lock to go away only when we actually delete the lock.  If the below resource_group delete fails for another reason then the lock we end up retrying for a long time instead of just failing right away,

What do you think?

Reply:
I added the retries here in order to allow deleting the resource group in case of the following cases:

1. Its take time for the lock to be actual removed
2. Its takes time for resource inside the rg to be actual destroyed (related to the discussion we had [here](https://github.com/redhat-cop/cloud.azure_ops/pull/73#discussion_r1606877886))

Solving the issue this way would solve also the same issue we might with other resources that might take longer time to delete.
In case user try to delete none empty resource group without indicating the "force" flag, it will wait 200 seconds, is it acceptable trade-off?

Any suggestion how to solve the second point, in case we use your suggestion for the first one?

Reply:
Thanks for the explanation.  I think your solution is a good compromise considering.